File: LeetCode_70/main.cpp
/**
    LeetCode_70
    https://leetcode.com/problems/climbing-stairs/description/
    MatrixDeity, 2017.
*/

#include <iostream>

using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        if (n == 0 || n == 1)
            return 1;

        int a = 1, b = 2;
        for (int i = 2; i < n; ++i) {
            int tmp = a;
            a = b;
            b = tmp + a;
        }
        return b;
    }
};

int main() {
    return 0;
}

File: LeetCode_70/main.go
/**
  LeetCode_70
  https://leetcode.com/problems/climbing-stairs/
  MatrixDeity, 2022.
*/

package leetcode70

func climbStairs(n int) int {
	dp := make([]int, n+1)

	for i := range dp {
		if i < 3 {
			dp[i] = i
		} else {
			dp[i] = dp[i-1] + dp[i-2]
		}
	}

	return dp[n]
}
